File: com_tower/pkg/structure/structure.go
package structure

import (
	"github.com/ViniiSouza/maritime_flow/com_tower/pkg/utils"
)

type StructureType string

const (
	PlatformStructureType StructureType = "platform"
	CentralStructureType  StructureType = "central"
)

type slots struct {
	DocksQtt    int `json:"docks_qtt"`
	HelipadsQtt int `json:"helipads_qtt"`
}

type Structure struct {
	Latitude  float64 `json:"latitude" db:"latitude"`
	Longitude float64 `json:"longitude" db:"longitude"`
	Slots     slots   `json:"slots"`
}

type Platform struct {
	Structure
	UUID utils.UUID `json:"platform_uuid" db:"id"`
}

type Central struct {
	Structure
	UUID utils.UUID `json:"central_uuid" db:"id"`
}

type Structures struct {
	Platforms []Platform `json:"platforms"`
	Centrals  []Central  `json:"centrals"`
}

File: mobility_core/MobilityCore/MobilityCore.Shared/Models/Structure.cs
using System.Text.Json.Serialization;

namespace MobilityCore.Shared.Models;

public class Structure
{
    [JsonPropertyName("central_uuid")]
    public string CentralUuid { get; set; } = string.Empty;
    
    [JsonPropertyName("platform_uuid")]
    public string PlatformUuid { get; set; } = string.Empty;
    
    [JsonPropertyName("latitude")]
    public double Latitude { get; set; }
    
    [JsonPropertyName("longitude")]
    public double Longitude { get; set; }
    
    [JsonPropertyName("slots")]
    public Slots Slots { get; set; } = new();
}

public class Slots
{
    [JsonPropertyName("docks_qtt")]
    public int DocksQtt { get; set; }
    
    [JsonPropertyName("helipads_qtt")]
    public int HelipadsQtt { get; set; }
}

